cmd/main.go

```diff
@@ -179,15 +197,24 @@ func setupManager(flagVar *flags.FlagVar, cacheOptions cache.Options, scheme *ma
 		setupLog.Error(err, "unable to set up ready check")
 		os.Exit(1)
 	}
-	if flagVar.DropStoredVersion != "" {
-		go func(version string) {
-			dropStoredVersion(mgr, version)
-		}(flagVar.DropStoredVersion)
-	}
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
-		setupLog.Error(err, "problem running manager")
-		os.Exit(1)
+}
+
+func runKymaMetricsCleanup(kymaMetrics *metrics.KymaMetrics, kcpClient client.Client,
+	cleanupIntervalInMinutes int,
+) {
+	scheduler := gocron.NewScheduler(time.UTC)
+	_, scheduleErr := scheduler.Every(cleanupIntervalInMinutes).Minutes().Do(func() {
+		ctx, cancel := context.WithTimeout(context.Background(), metricCleanupTimeout)
+		defer cancel()
+		if err := kymaMetrics.CleanupNonExistingKymaCrsMetrics(ctx, kcpClient); err != nil {
+			setupLog.Info(fmt.Sprintf("failed to cleanup non existing kyma crs metrics, err: %s", err))
+		}
+	})
```

Comment:
Just an idea, and not sure if this is common in Go... When testing locally, I was wondering if there is a possibility to see that the job was run but failed to clean up the metric without raising an error. It may be worth adding an Info or Debug log at the end of this function indicating that a clean up run has completed.

Reply:
Basically just for being able to see that the job is alive, even though no metrics have been purged in a while